lib/CommandCall.js

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) International Business Machines Corp. 2019
+// All Rights Reserved
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+// associated documentation files (the "Software"), to deal in the Software without restriction,
+// including without limitation the rights to use, copy, modify, merge, publish, distribute,
+// sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in all copies or
+// substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+// NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+const {
+  iXmlNodeCmdOpen,
+  iXmlNodeCmdClose,
+  iXmlNodeQshOpen,
+  iXmlNodeQshClose,
+  iXmlNodeShOpen,
+  iXmlNodeShClose,
+} = require('./ixml');
+
+const availableCommands = ['sh', 'cl', 'qsh'];
+
+class CommandCall {
+  /**
+   * @description Creates a new CommandCall object
+   * @constructor
+   * @param {object} config
+   */
+  constructor(config) {
+    if (!config.command || !config.type) {
+      throw new Error('Please specify a command and type (cl, qsh, sh)');
+    }
+
+    if (!availableCommands.includes(config.type)) {
+      throw new Error(`Invalid command type (${config.type}). Valid types: (${availableCommands.join(', ')})`);
+    }
+
+    this.command = config.command;
+    this.type = config.type;
+    this.options = config.options || null;
+    this.xml = '';
+  }
+
+  /**
+   * @description returns the command in xml format
+   * @returns {string}
+   */
+  toXML() {
+    if (this.type === 'sh') {
+      if (this.options) {
+        this.xml = iXmlNodeShOpen(this.options.rows, this.options.hex, this.options.before,
+          this.options.after, this.options.error)
+              + this.command + iXmlNodeShClose();
+      } else {
+        this.xml = iXmlNodeShOpen('', '', '', '', '') + this.command + iXmlNodeShClose();
+      }
+    } else if (this.type === 'cl') {
+      const xexec = (this.command.indexOf('?') > 0) ? 'rexx' : 'cmd';
```

Comment:
`xexec`?

Reply:
weird naming should just be exec:


```text
* <cmd [exec='cmd|system|rexx'
      *       hex='on' 
      *       before='cc1/cc2/cc3/cc4'
      *       after='cc4/cc3/cc2/cc1'
      *       error='on|off|fast'
      *       ]>values (see IBM i *CMD)</cmd>
```

Reply:
`x<whatever>` is the naming convention of variables used in 0.x when creating XML. Its all throughout ixml.js, but agree that the `x` should be rooted out wherever it is found.

Reply:
Sounds good will change it to `exec`

Reply:
@kadler @markdirish 

renamed `xexec` -> `exec`